darts/utils/cross_validation.py

```diff
@@ -0,0 +1,129 @@
+"""
+Cross-validation functions
+--------------------------
+"""
+import numpy as np
+import pandas as pd
+
+from typing import Union, Callable, Optional
+
+from darts.timeseries import TimeSeries
+from darts.logging import get_logger, raise_if_not, raise_if
+from darts.models.forecasting_model import ForecastingModel
+import darts.metrics as metrics
+from darts.utils import get_index_at_point
+
+logger = get_logger(__name__)
+
+
+def generalized_rolling_origin_evaluation(series: TimeSeries,
```

Comment:
@LeoTafti Couldn't we replace this by the new backtest() method you're working on?

Reply:
Good point, they would indeed be almost identical.

The most notable difference I see is that `groe()` allows to specify a maximum number of prediction steps via `n_prediction_steps`. If we want to keep this (but do we ? I'd say no), I could move this functionality over to `backtest()`.

Reply:
I did some more thinking and thought that a nicer way would be to provide a `stop` parameter instead.

But this might actually not be needed at all: it's already possible (and takes only one line) to trim the end of the series first before calling `historical_forecasts()` or `backtest()` to achieve the same result.

Since `historical_forecasts()` and `backtest()` signatures are already long/complicated enough imo, I'd be for leaving this out. What do you think ?

Reply:
Yes, sounds good